Classes/Util.php

```diff
@@ -462,11 +462,23 @@ public static function initializeTsfe(
             // for certain situations we need to trick TSFE into granting us
             // access to the page in any case to make getPageAndRootline() work
             // see http://forge.typo3.org/issues/42122
-            $pageRecord = BackendUtility::getRecord('pages', $pageId, 'fe_group');
-            $groupListBackup = $GLOBALS['TSFE']->gr_list;
-            $GLOBALS['TSFE']->gr_list = $pageRecord['fe_group'];
+            $pageRecord = BackendUtility::getRecord('pages', $pageId, 'fe_group, pid');
 
             $GLOBALS['TSFE']->sys_page = GeneralUtility::makeInstance(PageRepository::class);
+            $groupListBackup = $GLOBALS['TSFE']->gr_list;
```

Comment:
What happens when $pageRecord['pid'] != 0 and $parent['extendToSubpages'] is never set then $GLOBALS['TSFE']->gr_list will never be set.